research-software-directory/RSD-as-a-service
# [Feat] embed-pages inside iframes

# Ability to embed RSD pages within an iFrame

Fixes #250

Changes proposed in this pull request:

*   [x] Be able to detect the URL param `?embed` to display or not the footer and header sections within the iFrame
*   [x] Display back button when embed option
*   [x] Works for Software pages and Project Pages (for now)

Note: this is a working example of functionality that works with any type of HTML

### Use example:

```html
<h2>Check our Software - Demo Embed pages</h2>
<iframe src="http://localhost/software?embed" title="RSD Software" width="100%" height="500" frameBorder="0"></iframe>

<h2>Check our Projects - Demo Embed pages</h2>
<iframe src="http://localhost/projects?embed" title="RSD Software" width="100%" height="500" frameBorder="0"></iframe>
```

How to test:

*   Run `make start`
*   open: http://localhost/embed\_example.html
*   you should be able to pick any software, search and go back to the main grid view after visiting a software of project

![](https://user-images.githubusercontent.com/4195550/175906127-61a0a5e2-637a-45eb-b88b-d418b590bdb4.png)

PR Checklist:
